src/Features/CSharp/Portable/QuickInfo/CSharpSemanticQuickInfoProvider.cs

```diff
@@ -42,5 +45,64 @@ protected override bool ShouldCheckPreviousToken(SyntaxToken token)
         {
             return !token.Parent.IsKind(SyntaxKind.XmlCrefAttribute);
         }
+
+        protected override ImmutableArray<TaggedText> TryGetNullabilityAnalysis(Workspace workspace, SemanticModel semanticModel, SyntaxToken token, CancellationToken cancellationToken)
+        {
+            // Anything less than C# 8 we just won't show anything, even if the compiler could theoretically give analysis
+            var parseOptions = (CSharpParseOptions)token.SyntaxTree.Options;
+            if (parseOptions.LanguageVersion < LanguageVersion.CSharp8)
+            {
+                return default;
+            }
+
+            if (!parseOptions.Features.ContainsKey("run-nullable-analysis"))
```

Comment:
what's this constant?

Reply:
Set with https://github.com/dotnet/roslyn/blob/master/src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectOptionsProcessor.cs#L204 , enables nullable analysis in the compiler. Results in SyntaxTree having nullable reference type information

Reply:
Compiler feature flag. Extracted existing uses into a constant.

Reply:
Ok.  so this is just temporary right, while nullable is behind a feature flag?  This would become a full fledged API at some point (and something is trackign that right)?

Reply:
Yes, there's two parts here:

1. The compiler flag that turns it on programmatically.
2. This flag that controls whether or not the semantic model is running nullable analysis and providing answers.

This was largely a backstop in case if:

1. The compiler could compile but the semantic model had issues, perhaps if there's invalid code.
2. If there were any perf problems because the nullable walker does more work than calls like SemanticModel.GetTypeInfo().

It's possible to do 2 even if 1 isn't being done and the compiler/IDE team have been dogfooding that as well.

@333fred Which bug is ultimately tracking removal of this?

Reply:
I don't think we have a bug explicitly tracking this, but I'm planning on a PR within the next day or two to make the semantic analysis conditional on whether we will be running the nullable walker at all, which I think should largely remove your need to check this flag.